src/v/cloud_topics/level_zero/stm/ctp_stm.cc

```diff
@@ -167,30 +167,36 @@ ss::future<> ctp_stm::do_apply(const model::record_batch& batch) {
     }
     vlog(_log.debug, "Applying record batch: {}", batch.header());
 
-    if (batch.header().type == model::record_batch_type::dl_placeholder) {
+    switch (batch.header().type) {
```

Comment:
why switch? the switch is not exhaustive and nested switch statement are usually less readable

Reply:
oh no reason i just assumed everyone dislikes if-statement-chains. happy to drop it.

> nested switch statement are usually less readable

oh yeh should probably factor that out too